Sources/_StringProcessing/Regex/Core.swift

```diff
@@ -55,6 +55,12 @@ public struct Regex<Output>: RegexComponent {
     // in libswiftParseRegexLiteral.
     self.init(ast: try! parseWithDelimiters(pattern))
   }
+  
+  @_spi(RegexBuilder)
+  public init<T>(castingCaptures regex: Regex<T>, to outputType: Output.Type = Output.self) {
```

Comment:
It's really about casting the "output", not just captures, right? Should it be called `castingOutput:`?